File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue<int> days_left;
    
    for (int i = 0; i < progresses.size(); i++) {
        int remain_progress = 100 - progresses[i];
        int days = remain_progress / speeds[i];
        
        if (remain_progress % speeds[i] != 0) {
            days += 1;
        }
        
        days_left.push(days);
    }
    
    int front_day = days_left.front();
    int count = 0;
    while (!days_left.empty()) {
        if (days_left.front() <= front_day) {
            count++;
            days_left.pop();
        } 
        else {
            answer.push_back(count);
            count = 0;
            front_day = days_left.front();
        }
    }
    
    answer.push_back(count);
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    
    days_left = [math.ceil((100 - p) / s) for p, s in zip(progresses, speeds)]

    front_day = days_left[0]
    count = 0
    for day in days_left:
        if day <= front_day:
            count += 1
        else:
            answer.append(count)
            count = 1      
            front_day = day

    answer.append(count)
    
    return answer